lib/serialDevice.js

```diff
@@ -266,11 +266,14 @@ module.exports = function(config) {
 
           // TODO: Dexcom-specific code to be removed,
           // once OSX El Capitan is updated to not require Dexcom restart
-          if((this.app.state._os === 'mac') && (deviceInfo.driverId === 'Dexcom') && (deviceInfo.path.match(deviceInfo.portPattern) === null)) {
-            return cb(new Error('You may also need to power off and on your Dexcom receiver. ' + deviceDebugInfo));
-          }
+          chrome.runtime.getPlatformInfo(function(platformInfo) {
+            if ((platformInfo.os === 'mac') && (deviceInfo.driverId === 'Dexcom') &&
+              (deviceInfo.path.match(deviceInfo.portPattern) === null)) {
+              return cb(new Error('You may also need to power off and on your Dexcom receiver. ' + deviceDebugInfo));
+            }
 
-          cb(new Error('Could not connect to a matching device port: ' + deviceDebugInfo));
+            cb(new Error('Could not connect to a matching device port: ' + deviceDebugInfo));
+          });
```

Comment:
OK, yes, you're right - `getPlatformInfo` is asynchronous and we still get the desired result.

Basically we're waiting for Apple to fix their El Capitan USB stack, then we can remove the Dexcom-specific error. Or at least until most of our user base have the Tidepool Mac USB driver installed.